library/cwm/src/lib/cwm/tabs.rb

```diff
@@ -1,180 +1,76 @@
-    abstract_method :contents
-    # @return [String] label defines name of tab header
-    abstract_method :label
+require "cwm/page"
+require "cwm/pager"
 
-    # @return [WidgetHash]
-    def cwm_definition
-      super.merge(
-        "widgets"       => cwm_widgets,
-        "custom_widget" => Yast::CWM.PrepareDialog(cwm_contents, cwm_widgets)
-      )
-    end
-
-    # get cwm style of widget definitions
-    # @note internal api only used as gate to communicate with CWM
-    # @return [Array<WidgetHash>]
-    def cwm_widgets
-      return @cwm_widgets if @cwm_widgets
-
-      widgets = nested_widgets
-      names = widgets.map(&:widget_id)
-      definition = Hash[widgets.map { |w| [w.widget_id, w.cwm_definition] }]
-      @cwm_widgets = Yast::CWM.CreateWidgets(names, definition)
-    end
-
-    # help that is result of used widget helps.
-    # If overwritting, do not forget to use `super`, otherwise widget helps will
-    # be missing
-    def help
-      Yast::CWM.MergeHelps(nested_widgets.map(&:cwm_definition))
-    end
-  end
+module CWM
+  # Compatibility alias.
+  # (Tab and Tabs were here first, later Page and Pager
+  # were abstracted to allow for TreePager)
+  Tab = Page
 
-  # useful to have tabs as widget. It contained {CWM::Tab} with its content
   # @see examples/object_api_tabs.rb
-  # FIXME: Make this a subclass of {Pager}
-  class Tabs < CustomWidget
-    # @param [Array<CWM::Tab>] tabs to be shown
-    def initialize(*tabs)
-      @tabs = tabs
-      @current_tab = nil
-      self.handle_all_events = true
-    end
-
-    # initializes tabs, show tab which is initial
-    def init
-      switch_tab(initial_tab_id)
-    end
-
-    def handle(event)
-      # pass it to content of tab at first, maybe something stop passing
-      res = Yast::CWM.handleWidgets(@current_tab.cwm_widgets, event)
-      return res if res
-
-      new_id = event["ID"]
-      tab = tab_for_id(new_id)
-
-      return nil unless tab
-
-      return nil if @current_tab.widget_id == new_id
-
-      unless validate
-        mark_tab(@current_tab)
-        return nil
+  class Tabs < Pager
+    # {Tabs} does not have instances:
+    # {Tabs.new} overrides {Class.new} and calls
+    # either {DumbTabPager.new} or {PushButtonTabPager.new}.
+    def self.new(*args)
+      if Yast::UI.HasSpecialWidget(:DumbTab)
+        DumbTabPager.new(*args)
+      else
+        PushButtonTabPager.new(*args)
       end
-
-      store_tab(@current_tab.widget_id)
-
-      switch_tab(new_id)
-
-      nil
-    end
-
-    # store content of current tab
-    def store
-      store_tab(@current_tab.widget_id)
-    end
-
-    # validates current tab
-    def validate
-      Yast::CWM.validateWidgets(@current_tab.cwm_definition["widgets"], "ID" => @current_tab.widget_id)
     end
+  end
 
-    def help
-      @current_tab ? @current_tab.help : ""
-    end
+  # A {Pager} for the GUI, using the DumbTab widget
```

Comment:
I really recommend to add here note, to not use it directly. And use Tabs instead, which smartly decide which tabs use.